reference/test/matrix/ell_kernels.cpp

```diff
@@ -0,0 +1,346 @@
+        v[0] = 1.0;
+        v[1] = 5.0;
+        v[0 + 1 * padding] = 3.0;
+        v[0 + 2 * padding] = 2.0;
+    }
+
+    std::shared_ptr<const gko::Executor> exec;
+    std::unique_ptr<Mtx> mtx1;
+    std::unique_ptr<Mtx> mtx2;
+};
+
+
+TEST_F(Ell, AppliesToDenseVector)
+{
+    auto x = gko::initialize<Vec>({2.0, 1.0, 4.0}, exec);
+    auto y = Vec::create(exec, 2, 1, 1);
+
+    mtx1->apply(x.get(), y.get());
+
+    EXPECT_EQ(y->at(0), 13.0);
+    EXPECT_EQ(y->at(1), 5.0);
+}
+
+
+TEST_F(Ell, AppliesToDenseMatrix)
+{
+    // clang-format off
+    auto x = gko::initialize<Vec>(
+        {{2.0, 3.0},
+         {1.0, -1.5},
+         {4.0, 2.5}}, exec);
+    // clang-format on
+    auto y = Vec::create(exec, 2, 2, 2);
+
+    mtx1->apply(x.get(), y.get());
+
+    EXPECT_EQ(y->at(0, 0), 13.0);
+    EXPECT_EQ(y->at(1, 0), 5.0);
+    EXPECT_EQ(y->at(0, 1), 3.5);
+    EXPECT_EQ(y->at(1, 1), -7.5);
+}
+
+
+TEST_F(Ell, AppliesLinearCombinationToDenseVector)
+{
+    auto alpha = gko::initialize<Vec>({-1.0}, exec);
+    auto beta = gko::initialize<Vec>({2.0}, exec);
+    auto x = gko::initialize<Vec>({2.0, 1.0, 4.0}, exec);
+    auto y = gko::initialize<Vec>({1.0, 2.0}, exec);
+
+    mtx1->apply(alpha.get(), x.get(), beta.get(), y.get());
+
+    EXPECT_EQ(y->at(0), -11.0);
+    EXPECT_EQ(y->at(1), -1.0);
+}
+
+
+TEST_F(Ell, AppliesLinearCombinationToDenseMatrix)
+{
+    auto alpha = gko::initialize<Vec>({-1.0}, exec);
+    auto beta = gko::initialize<Vec>({2.0}, exec);
+    // clang-format off
+    auto x = gko::initialize<Vec>(
+        {{2.0, 3.0},
+         {1.0, -1.5},
+         {4.0, 2.5}}, exec);
+    auto y = gko::initialize<Vec>(
+        {{1.0, 0.5},
+         {2.0, -1.5}}, exec);
+    // clang-format on
+    mtx1->apply(alpha.get(), x.get(), beta.get(), y.get());
+
+    EXPECT_EQ(y->at(0, 0), -11.0);
+    EXPECT_EQ(y->at(1, 0), -1.0);
+    EXPECT_EQ(y->at(0, 1), -2.5);
+    EXPECT_EQ(y->at(1, 1), 4.5);
+}
+
+
+TEST_F(Ell, ApplyFailsOnWrongInnerDimension)
+{
+    auto x = Vec::create(exec, 2, 2, 2);
+    auto y = Vec::create(exec, 2, 2, 2);
+
+    ASSERT_THROW(mtx1->apply(x.get(), y.get()), gko::DimensionMismatch);
+}
+
+
+TEST_F(Ell, ApplyFailsOnWrongNumberOfRows)
+{
+    auto x = Vec::create(exec, 3, 2, 2);
+    auto y = Vec::create(exec, 3, 2, 2);
+
+    ASSERT_THROW(mtx1->apply(x.get(), y.get()), gko::DimensionMismatch);
+}
+
+
+TEST_F(Ell, ApplyFailsOnWrongNumberOfCols)
+{
+    auto x = Vec::create(exec, 3, 3, 2);
+    auto y = Vec::create(exec, 2, 2, 2);
+
+    ASSERT_THROW(mtx1->apply(x.get(), y.get()), gko::DimensionMismatch);
+}
+
+
+TEST_F(Ell, ConvertsToDense)
+{
+    auto dense_mtx = gko::matrix::Dense<>::create(mtx1->get_executor());
+    // clang-format off
+    auto dense_other = gko::initialize<gko::matrix::Dense<>>(
+        4, {{1.0, 3.0, 2.0},
+            {0.0, 5.0, 0.0}}, exec);
+    // clang-format on
+    mtx1->convert_to(dense_mtx.get());
+
+    ASSERT_MTX_NEAR(dense_mtx, dense_other, 0.0);
```

Comment:
No need for `dense_other`, you can pass the initializer list explicitly as shown in my other comments.

Reply:
Fix it.